lib/ansible/modules/cloud/amazon/s3.py

```diff
@@ -527,7 +527,11 @@ def main():
     # otherwise the connection will fail. See https://github.com/boto/boto/issues/2836
     # for more details.
     if '.' in bucket:
-        aws_connect_kwargs['calling_format'] = OrdinaryCallingFormat()
+        # don't set OrdinaryCallingFormat for fakes3 or rgw since it is used by default
+        if s3_url and (rgw or is_fakes3(s3_url)):
+            pass
+        else:
+            aws_connect_kwargs['calling_format'] = OrdinaryCallingFormat()
```

Comment:
This does fix the problem I was having, but it works because of an undocumented dependency between the set of conditions listed here and those listed at the top of `get_s3_connection`. It would be good to make a note that these conditions must cover all cases in `get_s3_connection` (I think in this case that they do cover everything provided that `is_fakes3(s3_url)` cannot be true unless `s3_url` is also true).